samples/DirectX/D3D12/HelloConstBuffer12.cs

```diff
@@ -0,0 +1,807 @@
+                            _constantBuffer->Unmap(0, null);
+                        }
+                    }
+                }
+
+                // Create synchronization objects and wait until assets have been uploaded to the GPU.
+                {
+                    fixed (ID3D12Fence** fence = &_fence)
+                    {
+                        iid = IID_ID3D12Fence;
+                        ThrowIfFailed(nameof(ID3D12Device.CreateFence), _device->CreateFence(0, D3D12_FENCE_FLAG_NONE, &iid, (void**)fence));
+                        _fenceValue = 1;
+                    }
+
+                    // Create an event handle to use for frame synchronization.
+                    _fenceEvent = CreateEventW(lpEventAttributes: null, bManualReset: FALSE, bInitialState: FALSE, lpName: null);
+                    if (_fenceEvent == null)
+                    {
+                        var hr = Marshal.GetHRForLastWin32Error();
+                        Marshal.ThrowExceptionForHR(hr);
+                    }
+
+                    // Wait for the command list to execute; we are reusing the same command
+                    // list in our main loop but for now, we just want to wait for setup to
+                    // complete before continuing.
+                    WaitForPreviousFrame();
+                }
+            }
+            finally
+            {
+                if (signature != null)
+                {
+                    signature->Release();
+                }
+
+                if (error != null)
+                {
+                    error->Release();
+                }
+
+                if (vertexShader != null)
+                {
+                    vertexShader->Release();
+                }
+
+                if (pixelShader != null)
+                {
+                    pixelShader->Release();
+                }
+            }
+        }
+
+        /// <summary> Fill the command list with all the render commands and dependent state. </summary>
+        private void PopulateCommandList()
+        {
+            // Command list allocators can only be reset when the associated
+            // command lists have finished execution on the GPU; apps should use
+            // fences to determine GPU execution progress.
+            ThrowIfFailed(nameof(ID3D12CommandAllocator.Reset), _commandAllocator->Reset());
+
+            // However, when ExecuteCommandList() is called on a particular command
+            // list, that command list can then be reset at any time and must be before
+            // re-recording.
+            ThrowIfFailed(nameof(ID3D12GraphicsCommandList.Reset), _commandList->Reset(_commandAllocator, _pipelineState));
+
+            // Set necessary state.
+            _commandList->SetGraphicsRootSignature(_rootSignature);
+
+            var pHeap = _cbvHeap;
+            _commandList->SetDescriptorHeaps(1, &pHeap);
```

Comment:
This should mirror native and match the port styling used elsewhere in the sample:
```suggestion
            const uint HeapsCount = 1;
            var ppHeaps = stackalloc ID3D12DescriptorHeap*[HeapsCount] {
                _cbvHeap
            };
            _commandList->SetDescriptorHeaps(HeapsCount, ppHeaps);
```